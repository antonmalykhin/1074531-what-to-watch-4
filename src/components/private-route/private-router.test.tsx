import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import history from '../../history';
import PrivateRouter from './private-route';

it(`PrivateRouter render`, () => {
  const mockStore = configureStore([]);

  const store = mockStore({
    APP: {
      currentYear: 2020,
      showedFilms: 8,
      isLoading: true
    },
    DATA: {
      activeFilter: `All genres`,
      comments: [],
      favoriteFilms: [],
      films: [],
      filters: [],
      isCommentSend: true,
      promoFilm: {}
    },
    USER: {
      authorizationStatus: `AUTH`,
      user: {}
    }
  });

  const tree = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <PrivateRouter
            exact
            path={`/login`}
            authorizationStatus={`AUTH`}
            render={() => {
              return <div />;
            }}
          />
        </Provider>
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
