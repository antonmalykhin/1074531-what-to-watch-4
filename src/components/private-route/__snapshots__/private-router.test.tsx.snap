// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRouter render 1`] = `null`;
